mk/prometheus.mk

```diff
@@ -0,0 +1,51 @@
+##
+# This makefile provide rules to start / stop a
+# local prometheus service.
+#
+# Variables:
+#   PROMETHEUS_VERSION
+#   PROMETHEYS_CONFIG
+#
+# See the container tags into the link below:
+#   https://hub.docker.com/r/prom/prometheus/tags
+#
+# See also the prometheus documentation at:
+#   https://prometheus.io/docs/introduction/overview/
+##
+
+PROMETHEUS_VERSION ?= v2.40.2
+PROMETHEUS_CONFIG ?= $(PROJECT_DIR)/configs/prometheus.yaml
+PROMETHEUS_UI_PORT ?= 9090
+export PROMETHEUS_UI_PORT
+export PROMETHEUS_CONFIG
+export PROMETHEUS_VERSION
+
+.PHONY: prometheus-up
+prometheus-up: ## Start prometheus service (local access at http://localhost:9090)
+	$(DOCKER) volume inspect prometheus &> /dev/null || $(DOCKER) volume create prometheus
+	$(DOCKER) container inspect prometheus &> /dev/null || \
+	$(DOCKER) run -d \
+	  --name prometheus \
+	  --restart always \
```

Comment:
is this container stoppable, even through Docker, if `--restart` is set to `always`? Reading here, it seems like it isn't https://docs.docker.com/engine/reference/commandline/run/#restart-policies---restart

Reply:
thanks mate! I have seen the prometheus container was not removed when calling `make prometheus-down`. I have updated both rules `prometheus-up` to remove `--restart always` (I added this to trace some situations when spinning up the container) and added the `--rm` option to remove the container when it is stopped.

You could need to run `make prometheus-clean`; if after run the above you see the prometheus container when running `podman container ls -a`, then run `podman container rm prometheus`. After this, and rebasing with the changes into the make rules, the expected lifecycle for the container using make rules should works properly (checked with podman, I have to check with docker).